src/Nox.Types/Types/CurrencyCode3/CurrencyCode3.cs

```diff
@@ -57,7 +38,7 @@ internal override ValidationResult Validate()
     {
         var result = base.Validate();
 
-        if (!_currencyCodes.Contains(Value))
+        if (!Enum.TryParse<CurrencyCode>(Value, out _))
```

Comment:
for performance reasons  in the static constructor build an hashset of valid string using the CurrencyCode Enum 

we should measure the difference, between Enum.TryParse and the hashset contains

Reply:
Suggestion: We can have an EnumExtension for the conversion of Enum values into HashSet<string> because we should be able to reuse it in another place

Reply:
Yes, HashSet should be faster because of constant access time, since we don't have many entries.
Changes have been applied accordingly.